File: C#/把鼠标当画笔/demo3.cs
using OpenCvSharp;
using System;

namespace demo
{
    /// <summary>
    /// 通过鼠标的左键和右键输出坐标
    /// </summary>
    public class demo3
    {

        public static Mat img = new Mat();

        // 鼠标左右键回调函数
        public static void onMouse(MouseEventTypes @event, int x, int y, MouseEventFlags flags, IntPtr userData)
        {
            if (@event == MouseEventTypes.LButtonDown)
            {
                Console.WriteLine($"{x},{y}");
            }
            if (@event == MouseEventTypes.RButtonDown)
            {
                Vec3b vec = img.Get<Vec3b>(y, x);
                byte blue = vec.Item0;
                byte green = vec.Item1;
                byte red = vec.Item2;
                Console.WriteLine($"{red},{green},{blue}");
                string strRGB = $"{red},{green},{blue}";

                var font = HersheyFonts.HersheySimplex;
                Cv2.PutText(img, strRGB, new Point(x, y), font, 1, new Scalar(255, 255, 255), 2);
                Cv2.ImShow("original", img);
            }
        }

        public static void Run()
        {
            img = Cv2.ImRead("../../../images/messi5.jpg");
            Cv2.ImShow("original", img);

            Cv2.NamedWindow("original");
            Cv2.SetMouseCallback("original", new MouseCallback(onMouse));
            Cv2.WaitKey(0);
            Console.Read();
        }
    }
}

File: Python/把鼠标当画笔/demo3.py
# OpenCv版本：opencv-python 4.6.0.66
# 内容：把鼠标当画笔
# 博客：http://www.bilibili996.com/Course?id=3993445000251
# 作者：高仁宝
# 时间：2023.11

import cv2


def click_event(event, x, y, flags, param):
    if event == cv2.EVENT_LBUTTONDOWN:
        print(x, y)

    if event == cv2.EVENT_RBUTTONDOWN:
        red = img[y, x, 2]
        blue = img[y, x, 0]
        green = img[y, x, 1]
        print(red, green, blue)

        strRGB = str(red) + "," + str(green) + "," + str(blue)
        font = cv2.FONT_HERSHEY_SIMPLEX
        cv2.putText(img, strRGB, (x, y), font, 1, (255, 255, 255), 2)
        cv2.imshow('original', img)


img = cv2.imread('../images/messi5.jpg')
cv2.imshow('original', img)

cv2.setMouseCallback("original", click_event)
cv2.waitKey(0)
cv2.imshow('putText.jpg', img)

cv2.destroyAllWindows()
